plugins/inputs/spectrum_scale/spectrum_scale_test.go

```diff
@@ -0,0 +1,274 @@
+		"time":              int(1427920124),
+		"time_microseconds": int(292780),
+		"write_bytes":       int(0),
+		"write_calls":       int(0),
+		"write_req":         int(0),
+	}
+
+	acc.AssertContainsTaggedFields(t, "spectrum_scale", fields, tags)
+}
+
+func verifySensorNsdds(t *testing.T, socketPath string) {
+
+	gfis := &Spectrum_Scale{
+		Sensors:        []string{"nsd_ds"},
+		SocketLocation: socketPath,
+	}
+
+	var acc testutil.Accumulator
+
+	err := gfis.Gather(&acc)
+	require.NoError(t, err)
+
+	tags := map[string]string{
+		"daemon_node_name": "node1.example.com",
+		"ipaddr":           "10.10.10.1",
+		"device":           "rg_node1_rg1",
+		"disk":             "rg_node1_rg1",
+		"sensor":           "nsd_ds",
+	}
+
+	fields := map[string]interface{}{
+		"longest_queued_time":  float64(0.141563),
+		"longest_wait_time":    float64(0.141563),
+		"lqtpb":                float64(0.000609),
+		"ltwpb":                float64(0.000609),
+		"rc":                   int(0),
+		"read_bytes":           int(563132928),
+		"read_calls":           int(138448),
+		"shortest_queued_time": float64(0.000145),
+		"shortest_wait_time":   float64(0.000145),
+		"sqtpb":                float64(0.000211),
+		"stwpb":                float64(0.000211),
+		"time":                 int(1427920124),
+		"time_microseconds":    int(292780),
+		"total_queued_time":    float64(3.652243),
+		"total_wait_time":      float64(3.652243),
+		"write_bytes":          int(202620928),
+		"write_calls":          int(9406),
+	}
+
+	acc.AssertContainsTaggedFields(t, "spectrum_scale", fields, tags)
+}
+
+func verifySensorFis(t *testing.T, socketPath string) {
+
+	gfis := &Spectrum_Scale{
+		Sensors:        []string{"fis"},
+		SocketLocation: socketPath,
+	}
+
+	var acc testutil.Accumulator
+
+	err := gfis.Gather(&acc)
+	require.NoError(t, err)
+
+	tags := map[string]string{
+		"cluster":          "cluster.example.com",
+		"daemon_node_name": "node1.example.com",
+		"filesystem":       "example",
+		"ipaddr":           "10.10.10.1",
+		"disk":             "2",
+		"sensor":           "fis",
+	}
+	fields := map[string]interface{}{
+		"close":             int(6631),
+		"inode_update":      int(1493),
+		"open":              int(6631),
+		"rc":                int(0),
+		"read_bytes_disk":   int(0),
+		"readdir":           int(6630),
+		"time":              int(1427920124),
+		"time_microseconds": int(292780),
+		"write_bytes":       int(0),
+		"write_req":         int(0),
+		"read_req":          int(0),
+	}
+
+	acc.AssertContainsTaggedFields(t, "spectrum_scale", fields, tags)
+}
+
+func TestGather(t *testing.T) {
+
+	// setup
+	tempdir := os.TempDir() + "/telegraf/"
+	socketName := "mmpmonSocket"
+
+	err := os.MkdirAll(tempdir, 0755)
+	require.NoError(t, err)
+
+	socketPath := tempdir + socketName
+
+	t.Logf("socketPath: %s", socketPath)
+
+	go socketHelper(t, socketPath)
+
+	// Initialize "server"
+	time.Sleep(2 * time.Second)
+
+	verifySensorGfis(t, socketPath)
```

Comment:
I'd prefer to see this structured as a table driven test instead of these three verify functions. https://github.com/golang/go/wiki/TableDrivenTests https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Table tests make the test easier to read because the input data and output data is in the same place and separate from the test code.  They're also easier to extend.